### PR TITLE
Remove NODE_ENV as it feels redundant 

### LINKED ISSUE #7: Deployment to now
Hey, thanks for this future tech. 

I am getting this under deployment to zeit/now: 

```javascript
> /home/nowuser/src/bin/server.js:40
>     const clientStats = stats.toJson().children[0]
>                              ^
> TypeError: Cannot read property 'toJson' of undefined
```

### PR DESCRIPTION
**Pull request by dawidvdh**
After #11 and #12 its starting to feel redundant to have `NODE_ENV` in the env file as webpack has separate scripts to define `production` for script output and the only role that `NODE_ENV` plays currently is running the server in either `production` or `development` mode. So shouldn't we rather just have separate scripts that get triggered via `npm`?

Its an alternative to #11 and #12 that resolves #7 .

### CODE REVIEW

File: bin/server.dev.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+const compiler = webpack([clientConfig, serverConfig])
+const clientCompiler = compiler.compilers[0]
+const options = { publicPath, stats: { colors: true } }
+
+app.use(webpackDevMiddleware(compiler, options))
+app.use(webpackHotMiddleware(clientCompiler))
+app.use(webpackHotServerMiddleware(compiler))
+
+compiler.plugin('done', done)
```
Comment (rocbear) [inline bin/server.dev.js]: @dawidvdh How about getting rid of `function done() { ...` and moving the callback inline now that we're only using it once?

Comment (dawidvdh) [inline bin/server.dev.js]: Yes totally agree, done.

**Comment by dawidvdh** [comment]
To me it this make more sense then #11 and I have seen this in practice in a few boilerplates but i hear you, How do we do this survey of a sorts? I actually just also really want to resolve #7 so maybe  we should merge #11 for the moment while we figure this out?

**Comment by dawidvdh** [comment]
However if we do only merge in #11 then we'll still experience the problem of `NODE_ENV=production` being in the .env file so `now` will not install `devDependencies` so then actually #11 wont resolve #7 unless paired with #12

**Comment by rocbear** [comment]
@dawidvdh alternative is moving `webpack` to `dependencies`? 

Either way, in the interest of playing nice with various other environments and hosting services, I think this is the safer approach for now. Also:

> I have seen this in practice in a few boilerplates

is enough of a study for the time being 😄 

I'm happy for you to merge this 👍

**Comment by dawidvdh** [comment]
Haha fair enough, cool all merged in!

**Pull request merged**